pandas/tests/io/parser/test_read_fwf.py

```diff
@@ -206,15 +206,17 @@ def test_fwf_compression(self):
         expected = read_fwf(StringIO(data), widths=widths, names=names)
         if compat.PY3:
             data = bytes(data, encoding='utf-8')
-        comps = [('gzip', gzip.GzipFile), ('bz2', bz2.BZ2File)]
-        for comp_name, compresser in comps:
-            with tm.ensure_clean() as path:
-                tmp = compresser(path, mode='wb')
-                tmp.write(data)
-                tmp.close()
-                result = read_fwf(path, widths=widths, names=names,
-                                  compression=comp_name)
-                tm.assert_frame_equal(result, expected)
+        comps = [('gzip', 'gz', gzip.GzipFile), ('bz2', 'bz2', bz2.BZ2File)]
+        for comp_name, extension, compresser in comps:
+            for kwargs in [
+                    {'compression': comp_name}, {'compression': 'infer'}, {}]:
```

Comment:
this needs to be parameterized.

@gfyoung don't we have functions to help compression testing along?